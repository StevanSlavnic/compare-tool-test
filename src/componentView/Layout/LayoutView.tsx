import React from 'react'

import { withProducts } from 'context/Products/ProductsConsumer'
import { CompareProvider } from 'context'

import { ProductsSideBar, ProductsList } from 'componentContainer'
import { HeaderView, FooterView } from 'componentView'

import { LayoutViewWrapper } from './Styled/LayoutViewStyled'

interface LayoutViewProps {
  productsContext: {
    state: {
      products: []
    }
    actions: {
      hideProduct: (id: string) => void
    }
  }
}

const LayoutView = (props: LayoutViewProps) => {
  const {
    productsContext: {
      state: { products },
      actions: { hideProduct }
    }
  } = props

  if (!products.length) {
    return 'Loading...'
  }

  return (
    <>
      <HeaderView products={products} />
      <CompareProvider products={products}>
        <LayoutViewWrapper>
          <ProductsSideBar products={products} hideProduct={hideProduct} />
          <ProductsList products={products} />
        </LayoutViewWrapper>
      </CompareProvider>
      <FooterView />
    </>
  )
}

export default withProducts(LayoutView)
