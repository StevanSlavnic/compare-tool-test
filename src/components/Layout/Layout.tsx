import React from 'react'

import { Header, Footer } from 'components'

interface LayoutProps {
  children: JSX.Element | JSX.Element[]
}

export const Layout = (props: LayoutProps) => {
  const { children } = props

  return (
    <>
      <Header />
      <main>{children}</main>
      <Footer />
    </>
  )
}
