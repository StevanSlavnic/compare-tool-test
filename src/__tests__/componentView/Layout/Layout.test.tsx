import React from 'react'
import { shallow } from 'enzyme'
import Layout from 'componentView/Layout/LayoutView'

it('renders Layout', () => {
  const component = shallow(<Layout />)
  expect(component).toMatchSnapshot()
})
