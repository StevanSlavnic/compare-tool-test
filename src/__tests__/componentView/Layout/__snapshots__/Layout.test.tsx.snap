// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Layout 1`] = `ShallowWrapper {}`;
